resources/protos/commit.proto

```diff
@@ -1,22 +1,33 @@
-syntax = "proto2";
+// Copyright 2017 Sourcerer, Inc. All Rights Reserved.
 
-package tutorial;
+syntax = "proto3";
+
+package app;
 
 option java_package = "app";
 option java_outer_classname = "CommitProtos";
 
-message Flag {
-    string type;
-    string value;
+// Types of information that can be extracted from commits.
+enum EnumLabelTypes {
+    LOC_ADDED = 0; // Lines of code added.
+    LOC_DELETED = 1; // Lines of code deleted.
+    QUALITY = 2; // Quality commit.
+    LANGUAGE = 3; // Programming languages used.
+    TECHNOLOGY = 4; // Technologies used.
 }
 
-message Commit {
-    optional int32 id;
-    optional string email;
-
-    required string name = 1;
-    required int32 id = 2;
-    optional string email = 3;
+// Label is a specific extracted information about commit.
+message Label {
+    int32 type = 1; // Type of extracted information.
+    string value = 2; // Extracted information.
+}
 
-    repeated Flag flags;
-}
\ No newline at end of file
+// Commit information message.
+message Commit {
+    int32 id = 1; // Commit id (hashed initial id).
+    int32 repo_id = 2; // Possibly hashed initial id of first commit.
```

Comment:
Possibly? You can't write comments like that. What will happen if you'll forget it here, and then it will be a very mysterious comment. If you don't know how it's done, don't say it here. If you do know, make it clear what it is. Or you can put a todo:

```
int32 repo_id = 2;  // Repo id. TODO(anatoly): add details on how it is computed.
```